lib/passwordless.rb

```diff
@@ -4,11 +4,18 @@
 require "passwordless/errors"
 require "passwordless/engine"
 require "passwordless/url_safe_base_64_generator"
+require "passwordless/token_generator"
 
 # The main Passwordless module
 module Passwordless
   mattr_accessor(:default_from_address) { "CHANGE_ME@example.com" }
-  mattr_accessor(:token_generator) { UrlSafeBase64Generator.new }
+  mattr_accessor(:token_generator) {
+    Passwordless::TokenGenerator.new(
+      ActiveSupport::CachingKeyGenerator.new(ActiveSupport::KeyGenerator.new('secret')),
```

Comment:
I copied this [from devise](https://github.com/heartcombo/devise/blob/715192a7709a4c02127afb067e66230061b82cf2/lib/devise/rails.rb#L42). I'm not sure whether it's necessary for `OpenSSL::HMAC` to get a secret "key". If so, should we use rails' `SECRET_KEY_BASE` or whatever?

Reply:
I think I'd like to keep CachingKeyGenerator out of this block – maybe move it to a `Passwordless.digest(token)` method? Then use it in a setter on `Session` like:

```ruby
def token=(value)
  self.token_digest = Passwordless.digest(value)
end
```

I think using Rails' secret key is a good idea 👍